### DISCUSSION TITLE
compile wasm error

### DISCUSSION THREAD

**Discussion by HaoWenJohn** [bevyengine/bevy#2416] [Q&A] [answered]
I'm trying to compile example/3d/3d_scene into wasm, using 
"cargo build --example 3d_scene --target wasm32-unknown-unknown --no-default-features --features="bevy_winit,bevy_gilrs,bevy_gltf,render,bevy_wgpu"" ,which causing many errors like :
error[E0432]: unresolved import `web_sys::GpuTextureFormat`
   --> /home/zhw/.cargo/registry/src/github.com-ad4b62560953c88a/wgpu-0.9.0/src/backend/web.rs:459:9
    |
459 |     use web_sys::GpuTextureFormat as tf;
    |
error[E0433]: failed to resolve: could not find `GpuAddressMode` in `web_sys`
   --> /home/zhw/.cargo/registry/src/github.com-ad4b62560953c88a/wgpu-0.9.0/src/backend/web.rs:837:51
    |
837 |         wgt::AddressMode::ClampToEdge => web_sys::GpuAddressMode::ClampToEdge,
    |                                                   ^^^^^^^^^^^^^^ could not find `GpuAddressMode` in `web_sys`

this may cause by bevy_wgpu feature, how can i solve the errors?

**Comment by MinerSebas** [comment, marked as answer]
WGPU currently does not support WASM. https://github.com/bevyengine/bevy/issues/793
Instead, you can use https://github.com/mrk-its/bevy_webgl2 as the render backend.
For  a quicker setup for WASM you can use these two Templates:
- https://github.com/mrk-its/bevy_webgl2_app_template
- https://github.com/NiklasEi/bevy_game_template

**Comment by The-DevBlog** [comment]
Hit me up if you have any questions about bevy_webgl2! I just got done with a simple project using it and ran into a lot of issues I had to solve.
